codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/CodegenVisitor.java

```diff
@@ -79,8 +80,10 @@ final class CodegenVisitor extends ShapeVisitor.Default<Void> {
         LOGGER.info("Attempting to discover GoIntegration from the classpath...");
         ServiceLoader.load(GoIntegration.class, loader)
                 .forEach(integration -> {
-                    LOGGER.info(() -> "Adding GoIntegration: " + integration.getClass().getName());
-                    integrations.add(integration);
+                    if (integration.getArtifactType().equals(ArtifactType.CLIENT)) {
```

Comment:
Is there any kind of future refactoring that removes the need to add these kinds of conditionals inside of codegen?